tasking/pulp/tasking/tasks.py

```diff
@@ -190,8 +191,8 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         resource_id = ":".join((resource_type, resource_id))
         inner_task_id = str(uuid.uuid4())
         task_name = self.name
-        tag_list = kwargs.get('tags', [])
-        group_id = kwargs.get('group_id', None)
+        tag_list = async_kwargs.get('tags', [])
```

Comment:
I think tags and group_id should become explicit arguments and not expected to be pulled out of async_kwargs. This way async_kwargs will only store kwargs that will be sent to celery.